lib/src/Uncertainty/Algorithm/Sensitivity/SobolIndicesAlgorithmImplementation.cxx

```diff
@@ -0,0 +1,800 @@
+{
+  if (0 == varianceI_.getSize())
+  {
+    // Invoke internal method to compute first/ total order indices indices
+    varianceI_ = computeIndices(outputDesign_, varianceTI_);
+  }
+  if (0 == mergedFirstOrderIndices_.getDimension())
+  {
+    // Compute aggregate indices
+    mergedFirstOrderIndices_ = computeAggregatedIndices(varianceI_, varianceTI_, referenceVariance_, mergedTotalOrderIndices_);
+  }
+  // Indices computed
+  return mergedFirstOrderIndices_;
+}
+
+/* Aggregated total order indices accessor for multivariate samples */
+NumericalPoint SobolIndicesAlgorithmImplementation::getAggregatedTotalOrderIndices() const
+{
+  if (0 == varianceI_.getSize())
+  {
+    // Invoke internal method to compute first/ total order indices indices
+    varianceI_ = computeIndices(outputDesign_, varianceTI_);
+  }
+  if (0 == mergedFirstOrderIndices_.getDimension())
+  {
+    // Compute aggregate indices
+    mergedFirstOrderIndices_ = computeAggregatedIndices(varianceI_, varianceTI_, referenceVariance_, mergedTotalOrderIndices_);
+  }
+  // Indices computed
+  return mergedTotalOrderIndices_;
+}
+
+// Setter for bootstrap number
+void SobolIndicesAlgorithmImplementation::setBootstrapSize(const UnsignedInteger nrBootstrap)
+{
+  if (nrBootstrap <= 0.0)
+    throw InvalidArgumentException(HERE) << "Number of bootstrap resampling size should be positive. Here, nrBootstrap=" << nrBootstrap;
+  nrBootstrap_ = nrBootstrap;
+  firstOrderIndiceInterval_ = Interval();
+  totalOrderIndiceInterval_ = Interval();
+}
+
+// Setter for bootstrap confidence level
+void SobolIndicesAlgorithmImplementation::setBootstrapConfidenceLevel(const NumericalScalar confidenceLevel)
+{
+  if ((confidenceLevel < 0.0) || (confidenceLevel >= 1.0))
+    throw InvalidArgumentException(HERE) << "Confidence level value should be in ]0,1[. Here, confidence level=" << confidenceLevel;
+  confidenceLevel_ = confidenceLevel;
+  firstOrderIndiceInterval_ = Interval();
+  totalOrderIndiceInterval_ = Interval();
+}
+
+/* String converter */
+String SobolIndicesAlgorithmImplementation::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << GetClassName()
+      << " name=" << getName();
+  return oss;
+}
+
+// Multiplication and sum of two NumericalSamples
+// TODO Write method in NumericalSample ?
+NumericalPoint SobolIndicesAlgorithmImplementation::computeSumDotSamples(const NumericalSample & x,
+                                                                         const NumericalSample & y) const
+{
+  // Internal method
+  // Suppose that samples have the same size, same dimension
+  const UnsignedInteger dimension(x.getDimension());
+  const UnsignedInteger size(x.getSize());
+
+  const NumericalScalar * xptr(&x[0][0]);
+  const NumericalScalar * yptr(&y[0][0]);
+
+
+  NumericalPoint value(dimension, 0.0);
+  for (UnsignedInteger i = 0; i < size; ++i)
+      for (UnsignedInteger j = 0; j < dimension; ++j, ++xptr, ++yptr)
+	  value[j] += (*xptr) * (*yptr);
```

Comment:
The problem is that this function is used inside a loop, and anything other than raw pointers is much slower because data are copied when using NP or NS.